test/unit/request-auth.test.js

```diff
@@ -4,123 +4,185 @@ var expect = require('chai').expect,
+                },
+                basic: {
+                    username: 'u', password: 'p'
+                },
+                type: 'noauth'
+            });
 
-    it('should be able to construct with multiple auths', function () {
-        var auth = new RequestAuth({
-            noauth: {
+            expect(auth.current()).to.eql({
                 foo: 'bar'
-            },
-            basic: {
+            });
+
+            auth.use('basic');
+            expect(auth.type).to.equal('basic');
+            expect(auth.current()).to.eql({
                 username: 'u', password: 'p'
-            },
-            type: 'noauth'
+            });
         });
 
-        expect(auth.current()).to.eql({
-            foo: 'bar'
-        });
+        it('should be able to return the parameters of the selected auth type', function () {
+            var auth = new RequestAuth({
+                noauth: {
+                    foo: 'bar'
+                },
+                type: 'noauth'
+            });
 
-        auth.use('basic');
-        expect(auth.type).to.equal('basic');
-        expect(auth.current()).to.eql({
-            username: 'u', password: 'p'
+            auth.use('noauth', {
+                noauth: {
+                    foo1: 'bar1'
+                },
+                type: 'noauth'
+            });
+            expect(auth.parameters().toObject()).to.eql({
+                foo: 'bar',
+                noauth: {
+                    'foo1': 'bar1'
+                },
+                type: 'noauth'
+            });
         });
     });
 
-    it('should be able to clear an auth that is currently selected', function () {
-        var auth = new RequestAuth({
-            noauth: {
-                foo: 'bar'
-            },
-            basic: {
-                username: 'u', password: 'p'
-            },
-            type: 'noauth'
-        });
+    describe('.update', function () {
+        it('should set the authentication type to be used by this item when options array is empty', function () {
+            var auth = new RequestAuth({
+                noauth: {
+                    foo: 'bar'
+                },
+                type: 'noauth'
+            });
 
-        expect(auth.current()).to.eql({
-            foo: 'bar'
+            auth.use('noauth', []);
+            expect(auth.type).to.equal('noauth');
+            auth.update(null);
+            expect(auth.parameters().toObject()).to.eql({
+                foo: 'bar'
+            });
         });
 
-        auth.clear('noauth');
+        it('should bail out when non-object options are passed to update', function () {
+            var auth = new RequestAuth({
+                noauth: {
+                    foo: 'bar'
+                },
+                type: 'noauth'
+            });
 
-        expect(auth.current()).to.eql({});
+            auth.update(null);
+            auth.update({});
+            expect(auth.parameters().toObject()).to.eql({
```

Comment:
It would make more sense to either break this test into multiple tests for each invalid value, or add expect after each `.update`.

We should also check for other invalid values like `string`, `numbers`, `boolean`, etc.

Reply:
Fixed